compiler/rustc_ty_utils/src/ty.rs

```diff
@@ -360,7 +360,8 @@ fn issue33140_self_ty(tcx: TyCtxt<'_>, def_id: DefId) -> Option<Ty<'_>> {
 
     let trait_ref = tcx
         .impl_trait_ref(def_id)
-        .unwrap_or_else(|| bug!("issue33140_self_ty called on inherent impl {:?}", def_id));
+        .unwrap_or_else(|| bug!("issue33140_self_ty called on inherent impl {:?}", def_id))
+        .skip_binder();
```

Comment:
`issue33140_self_ty` should be changed to return `Option<EarlyBinder<Ty<'_>>>` as well.

Might be nice to also do that in a separate PR if you're interested.